src/azure-cli-core/setup.py

```diff
@@ -52,7 +52,7 @@
     'humanfriendly~=10.0',
     'jmespath',
     'knack~=0.11.0',
-    'microsoft-security-utilities-secret-masker~=1.0.0b2',
+    'microsoft-security-utilities-secret-masker~=1.0.0b4',
```

Comment:
Why it is not in `requirements.txt`?

Reply:
It's `azure-cli-core`'s dependency, not `azure-cli`

Reply:
I see.
This change has no effect since `~=` will select the `1.0.0b4` by default.

Reply:
Right, this PR fixes tests after sdk upgrade as well

Reply:
> It's `azure-cli-core`'s dependency, not `azure-cli`

This is not accurate. `requirements.txt` contains requirements of both Azure CLI and Core. `requirements.txt` is used to have stable dependencies in MSI, ZIP, DEB, RPM packages.

If `microsoft-security-utilities-secret-masker` is required by Azure CLI Core, it should be included in `requirements.txt` too. `msal` is an example:

https://github.com/Azure/azure-cli/blob/1696f9fe0e31bb89c5eef0d1a1d73bf424cc4a9a/src/azure-cli-core/setup.py#L57

https://github.com/Azure/azure-cli/blob/cd9a1ae91ee55fac691a95fd3887a7538dce20c5/src/azure-cli/requirements.py3.windows.txt#L106

Reply:
I upvote adding some `azure-cli-core` dependencies in `requirements.txt`.

I have to manually add `microsoft-security-utilities-secret-masker` and `py-deviceid` to `requirements.txt` in: https://github.com/Azure/azure-cli-pr/pull/143#discussion_r2030667413